Watchman.DomainModel/Configuration/Services/ConfigurationService.cs

```diff
@@ -0,0 +1,141 @@
+            if (existingConfiguration == null)
+            {
+                await session.AddAsync(baseFormatConfigurationItem);
+            }
+            else
+            {
+                existingConfiguration.SetValue(baseFormatConfigurationItem.Value);
+                await session.AddOrUpdateAsync(existingConfiguration);
+            }
+            this.Refresh();
+        }
+
+        public async Task InitDefaultConfigurations()
+        {
+            var configurationsTypes = this._configurationTypesSearcher.ConfigurationTypes;
+            var configurations = configurationsTypes.Select(x =>
+            {
+                var conf = (IMappedConfiguration)Activator.CreateInstance(x, DEFAULT_SERVER_ID);
+                return this._configurationMapperService.MapIntoBaseFormat(conf);
+            });
+            using var session = this._sessionFactory.Create();
+            var existingConfigurations = session.Get<ConfigurationItem>().Where(x => x.ServerId == 0).ToList();
+            foreach (var configuration in configurations.Where(x => existingConfigurations.All(y => x.Name != y.Name)))
+            {
+                await session.AddAsync(configuration);
+            }
+            this.Refresh();
+        }
+
+        public void Refresh()
+        {
+            using var session = this._sessionFactory.Create();
+            var configurationItems = session.Get<ConfigurationItem>().ToList();
+            var mappedConfigurations = this._configurationMapperService.GetMappedConfigurations(configurationItems);
+            if (_configurationVersions != null)
+            {
+                var tasks = new List<Task>();
+                var changedConfigurationItems = configurationItems.Where(x => _configurationVersions.GetValueOrDefault(x.Id) != x.Version);
+                foreach (var changedConfiguration in changedConfigurationItems)
+                {
+                    var sameTypeMappedConfigurations = mappedConfigurations.FirstOrDefault(x => x.Key.Name == changedConfiguration.Name).Value;
+                    var mappedConfiguration = sameTypeMappedConfigurations[changedConfiguration.ServerId];
+                    var configurationChangesHandler = this.GetConfigurationChangesHandler(mappedConfiguration);
+                    if (configurationChangesHandler == null)
+                    {
+                        throw new NotImplementedException($"configurationChangesHandler for {changedConfiguration.Name} is not implemented");
+                    }
+                    var task = configurationChangesHandler.Handle(changedConfiguration.ServerId, mappedConfiguration);
+                    tasks.Add(task);
+                }
+                Task.WaitAll(tasks.ToArray());
+            }
+            _configurationVersions = configurationItems.ToDictionary(x => x.Id, x => x.Version);
+            //var changedConfigurations = mappedConfigurations.SelectMany(x =>
```

Comment:
a co to za potężne komentarze?
zostawiłeś przypadkiem czy celowo bo kiedyś można będzie z nich skorzystać?